js/data/buffer.js

```diff
@@ -83,7 +83,7 @@ Buffer.prototype.setVertexAttribPointers = function(gl, program) {
  * @param gl The WebGL context
  */
 Buffer.prototype.destroy = function(gl) {
-    if (this.buffer) {
+    if (gl && this.buffer) {
```

Comment:
A falsy value for `gl` indicates that we have manually destroyed the GL context. Tracking this information prevents "invalid GL context" errors in the runtime styling tests when asynchronous operations are run after the context has been destroyed. 

I will investigate some alternate designs today.